src/nanovg_pick.c

```diff
@@ -0,0 +1,1651 @@
+			}
+			i += 7;
+			break;
+		case NVG_CLOSE:
+			if (ps->points[(ps->npoints - 1) * 2] != start[0] ||
+				ps->points[(ps->npoints - 1) * 2 + 1] != start[1]) {
+				firstPoint = nvg__pickSceneAddPoints(ps, start, 1);
+				nvg__pickSubPathAddSegment(ps, psp, firstPoint - 1, NVG_LINETO, NVG_PICK_CORNER);
+			}
+			psp->closed = 1;
+			
+			i++;
+			break;
+		case NVG_WINDING:
+			psp->winding = (short)(int)commands[i+1];
+			if (psp->winding == NVG_HOLE)
+				hasHoles = 1;
+			i += 2;
+			break;
+		default:
+			i++;
+			break;
+		}
+	}
+	
+	pp->flags = (hasHoles ? NVG_PICK_HASHOLES : 0) | (forStroke ? NVG_PICK_STROKE : NVG_PICK_FILL);
+	pp->subPaths = psp;
+	pp->strokeWidth = state->strokeWidth * 0.5f;
+	pp->miterLimit = state->miterLimit;
+	pp->lineCap = (short)state->lineCap;
+	pp->lineJoin = (short)state->lineJoin;
+	
+	nvg__initBounds(totalBounds);
+
+	for (curpsp = psp; curpsp; curpsp = curpsp->next) {	
+		if (forStroke)
+			nvg__pickSubPathAddStrokeSupports(ps, curpsp, pp->strokeWidth, pp->lineCap, pp->lineJoin, pp->miterLimit);
+		else
+			nvg__pickSubPathAddFillSupports(ps, curpsp);
+		
+		segments = &ps->segments[curpsp->firstSegment];
+		nvg__initBounds(curpsp->bounds);
+		for (s = 0; s < curpsp->nsegments; ++s) {
+			seg = &segments[s];
+			NVG_PICK_DEBUG_BOUNDS(seg->bounds);		
+			nvg__unionBounds(curpsp->bounds, seg->bounds);
+		}
+
+		nvg__unionBounds(totalBounds, curpsp->bounds);
+	}
+	
+	// Store the scissor rect if present.
+	if (state->scissor.extent[0] != -1.0f) {	
+		// Use points storage to store the scissor data
+		float* scissor = NULL;
+		pp->scissor = nvg__pickSceneAddPoints(ps, NULL, 4);
+		scissor = &ps->points[pp->scissor*2];
+	
+		memcpy(scissor, state->scissor.xform, 6 * sizeof(float));
+		memcpy(scissor + 6, state->scissor.extent, 2 * sizeof(float));
+		
+		pp->flags |= NVG_PICK_SCISSOR;
+	}
+				
+	memcpy(pp->bounds, totalBounds, sizeof(float) * 4);
+	
+	return pp;
+}
+
+// Struct management
+
+static NVGpickPath* nvg__allocPickPath(NVGpickScene* ps)
+{
+	NVGpickPath* pp = ps->freePaths;
+	if (pp) {
+		ps->freePaths = pp->next;
+	} else {
+		pp = (NVGpickPath*)malloc(sizeof(NVGpickPath));		
+	}
+	memset(pp, 0, sizeof(NVGpickPath));
+	return pp;
+}
+
+// Return a pick path and any sub paths to the free lists
+static void nvg__returnPickPath(NVGpickScene* ps, NVGpickPath* pp)
+{
+	// Add all sub paths to the sub path free list.
+	// Finds the end of the path sub paths, links that to the current
+	// sub path free list head and replaces the head ptr with the 
+	// head path sub path entry.
+	NVGpickSubPath* psp = NULL;
+	for (psp = pp->subPaths; psp && psp->next; psp = psp->next) {
+	}
+
+	if (psp) {
+		psp->next = ps->freeSubPaths;
+		ps->freeSubPaths = pp->subPaths;
+	}
+	pp->subPaths = NULL;
+	
+	// Add the path to the path freelist
+	pp->next = ps->freePaths;
+	ps->freePaths = pp;	
+	if (pp->next == NULL)
+		ps->lastPath = pp;
+}
+
+static void nvg__deletePickPath(NVGpickPath* pp)
```

Comment:
Used only once, fold into `nvg__deletePickScene()`, reuse the name for return as per above.